File: src/Rotate Array/Rotate Array.cpp
class Solution {
	public:
		void rotate(int nums[], int n, int k) {
			if(!k || !(k%n) ) return;
			k = k%n;
			vector<int> temp(nums, nums+n);
			int start = n-k;
			for(int i = 0 ; i < n ; i++)
			{
				nums[i] = temp[start++];
				if(start == n) start = 0;
			}
		}
};

File: src/Rotate Array/Rotate Array.py
class Solution:
	# @param nums, a list of integer
	# @param k, num of steps
	# @return nothing, please modify the nums list in-place.
	def rotate(self, nums, k):
		n=len(nums)
		if not k or not k%n: return
		k%=n
		for index,c in enumerate( nums[-k:] + nums[:n-k] ):
			nums[index]=c 
